controllers/booking_controller.go

```diff
@@ -0,0 +1,182 @@
+// @Description Retrieve a list of upcoming bookings for the trainer who is currently logged in
+// @Tags bookings
+// @Accept json
+// @Produce json
+// @Security BearerAuth
+// @Success 200 {object} responses.GetBookingsResponse
+// @Failure 400 {object} responses.GetBookingsResponse
+// @Router /protected/bookings [GET]
+func GetBookings() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		username, err := tokens.ExtractTokenUsername(c)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, responses.GetBookingsResponse{
+				Status:  http.StatusBadRequest,
+				Message: err.Error(),
+			})
+		}
+		result, err := models.GetUpcomingBookingsForTrainer(username)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, responses.GetBookingsResponse{
+				Status:  http.StatusBadRequest,
+				Message: err.Error(),
+			})
+			return
+		}
+		c.JSON(http.StatusOK, responses.GetBookingsResponse{
+			Status:   http.StatusOK,
+			Message:  `success!`,
+			Bookings: result,
+		})
+	}
+}
+
+// @Summary Update a booking
+// @Description Update a booking of sepecified bookingId with the specified update input consist of status(pending/confirm/complete) and paymentStatus(pending/paid)
+// @Tags bookings
+// @Accept json
+// @Produce json
+// @Security BearerAuth
+// @Param json_in_ginContext body UpdateBookingForm true "put updateBookingForm details and pass to gin.Context"
+// @Success	200		{object}	responses.UpdateBookingResponse	"Successfully update booking"
+// @Failure	400		{object}	responses.UpdateBookingResponse	"Bad Request, missing filed of objectId or cannot find bookingObjectId"
+// @Router /protected/update-booking [post]
+func UpdateBooking() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var input UpdateBookingForm
+		if err := c.ShouldBindJSON(&input); err != nil {
+			c.JSON(http.StatusBadRequest, responses.UpdateBookingResponse{
+				Status:  http.StatusBadRequest,
+				Message: err.Error(),
+			})
+			return
+		}
+
+		err := models.UpdateBooking(input.BookingId, input.Status, input.PaymentStatus)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, responses.UpdateBookingResponse{
+				Status:  http.StatusBadRequest,
+				Message: `update failed ` + err.Error(),
+			})
+			return
+		}
+
+		c.JSON(http.StatusOK,
+			responses.UpdateBookingResponse{
+				Status:  http.StatusOK,
+				Message: `update booking success!`,
+			})
+	}
+}
+
+// @Summary Delete a booking
+// @Description Delete a booking with the specified bookingId
+// @Tags bookings
+// @Accept json
+// @Produce json
+// @Security BearerAuth
+// @Param json_in_ginContext body DeleteBookingForm true "put DeleteBookingForm details and pass to gin.Context"
+// @Success	200		{object}	responses.DeleteBookingResponse	"Successfully delete booking"
+// @Failure	400		{object}	responses.DeleteBookingResponse	"Bad Request, missing filed of objectId or cannot find bookingObjectId"
+// @Router /protected/delete-booking [post]
+func DeleteBooking() gin.HandlerFunc {
```

Comment:
Should this routes be a `DELETE` instead of a `POST`?

Reply:
Use this for now